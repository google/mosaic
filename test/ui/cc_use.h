struct A { char a; };
struct B { char a; };
namespace Foo {
    struct C { char a; };
}
