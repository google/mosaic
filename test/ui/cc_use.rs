cc_use!("cc_use.h", Foo, Baz, Bar);
