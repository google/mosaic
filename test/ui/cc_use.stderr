error: expected literal
  ┌─ test/ui/cc_use.rs:2:23
  │
2 │ cc_use!("cc_use.h" in libfoo, B);
  │                       ^^^^^^ expected literal

error: expected `>`
  ┌─ test/ui/cc_use.rs:4:16
  │
4 │ cc_use!(<cc_use.h>, A);
  │                ^ expected `>`

error: Try putting system paths inside quotes: "<foo/bar.h>"
  ┌─ test/ui/cc_use.rs:4:9
  │
4 │ cc_use!(<cc_use.h>, A);
  │         ^ Try putting system paths inside quotes: "<foo/bar.h>"

error: expected identifier
  ┌─ test/ui/cc_use.rs:5:21
  │
5 │ cc_use!("cc_use.h", in "libfoo", B);
  │                     ^^ expected identifier

error: expected `in` or `,`
  ┌─ test/ui/cc_use.rs:6:19
  │
6 │ cc_use!("cc_use.h"; B);
  │                   ^ expected `in` or `,`

error: expected `in` or `,`
  ┌─ test/ui/cc_use.rs:7:20
  │
7 │ cc_use!("cc_use.h" on "libfoo", B);
  │                    ^^ expected `in` or `,`

error: 'nonexistent.h' file not found
  ┌─ test/ui/cc_use.rs:2:10
  │
2 │ #include "nonexistent.h"
  │          ^^^^^^^^^^^^^^^

error: item not found: `Zed`
  ┌─ test/ui/cc_use.rs:1:24
  │
1 │ cc_use!("cc_use.h", A, Zed, B);
  │                        ^^^ this item could not be found

error: item not found: `Foo::D`
  ┌─ test/ui/cc_use.rs:3:44
  │
3 │ cc_use!("cc_use.h" in "libfoo", B, Foo::C, Foo::D, Bar::E);
  │                                            ^^^^^^ this item could not be found

error: item not found: `Bar::E`
  ┌─ test/ui/cc_use.rs:3:52
  │
3 │ cc_use!("cc_use.h" in "libfoo", B, Foo::C, Foo::D, Bar::E);
  │                                                    ^^^^^^ this item could not be found

error: item not found: `B`
  ┌─ test/ui/cc_use.rs:8:38
  │
8 │ cc_use!("nonexistent.h" in "libfoo", B);
  │                                      ^ this item could not be found
